src/timessquare/handlers/v1/handlers.py

```diff
@@ -10,58 +12,76 @@
     context_dependency,
 )
 
-from .models import Index, Page, PostPageRequest
+from .models import Index, Page, PageSummary, PostPageRequest
 
 __all__ = ["v1_router"]
 
-v1_router = APIRouter()
+v1_router = APIRouter(tags=["v1"])
 """FastAPI router for all external handlers."""
 
 
 @v1_router.get(
     "/",
-    description=(
-        "Metadata about the v1 REST API, including links to "
-        "documentation and endpoints."
-    ),
     response_model=Index,
     summary="V1 API metadata",
 )
 async def get_index(
     request: Request,
 ) -> Index:
+    """Get metadata about the v1 REST API, including links to documentation and
+    endpoints.
+    """
     metadata = get_metadata(
         package_name="times-square",
         application_name=config.name,
     )
-    return Index(metadata=metadata, api_docs=f"{request.url}docs")
+    doc_url = request.url.replace(path=f"/{config.name}/docs")
+    return Index(metadata=metadata, api_docs=str(doc_url))
 
 
 @v1_router.get(
     "/pages/{page}",
-    description=(
-        "Get metadata about a page resource, which models a webpage that is "
-        "rendered from a parameterized Jupyter Notebook."
-    ),
     response_model=Page,
     summary="Page metadata.",
     name="get_page",
 )
 async def get_page(
     page: str, context: RequestContext = Depends(context_dependency)
 ) -> Page:
+    """Get metadata about a page resource, which models a webpage that is
+    rendered from a parameterized Jupyter Notebook.
+    """
     page_service = context.page_service
-    page_domain = await page_service.get_page(page)
+    async with context.session.begin():
+        page_domain = await page_service.get_page(page)
 
-    context.response.headers["location"] = context.request.url_for(
-        "get_page", page=page_domain.name
-    )
-    return Page.from_domain(page=page_domain, request=context.request)
+        context.response.headers["location"] = context.request.url_for(
+            "get_page", page=page_domain.name
+        )
+        return Page.from_domain(page=page_domain, request=context.request)
```

Comment:
This is more superstition than anything, but I would move this outside of the context manager in the name of minimizing the amount of time you spend in a transaction, since it doesn't use the database and the transaction forces serialization on the database side and ideally should be as short as possible.